### PR TITLE
Remove some type parameters from MPIStateArray

### PR DESCRIPTION
**Pull request by jkozdon**

### CODE REVIEW

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
bors try

File: src/Arrays/MPIStateArrays.jl [hunk 1]
```diff
@@ -417,24 +405,25 @@ quadrature weights from a grid, thus this becomes an integral approximation.
...
   isempty(A.weights) && error("`weightedsum` requires weights")
 
-  T = eltype(A)
+  FT = eltype(A)
   states = SVector{length(states)}(states)
 
   C = @view A.data[:, states, A.realelems]
   w = @view A.weights[:, :, A.realelems]
-  init = zero(DoubleFloat{T})
+  init = zero(DoubleFloat{FT})
```
Comment (charleskawczynski) [inline src/Arrays/MPIStateArrays.jl]: Why is `DoubleFloat` used here? To maintain accuracy? Also, what happens if `FT` is higher precision than `DoubleFloat`?

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try

**Comment by jkozdon** [comment]
bors try

File: src/Arrays/MPIStateArrays.jl [hunk 2]
```diff
@@ -4,31 +4,28 @@ using DoubleFloats
 using LazyArrays
 using StaticArrays
 using GPUifyLoops
+import ..haspkg
```
Comment (charleskawczynski) [inline src/Arrays/MPIStateArrays.jl]:
```suggestion
using ..haspkg
```

Comment (jkozdon) [reply src/Arrays/MPIStateArrays.jl]: Doesn't work for me locally to do it this way.

Comment (charleskawczynski) [reply src/Arrays/MPIStateArrays.jl]: That's odd, why is that? @simonbyrne

**Comment by jkozdon** [comment]
@kpamnany and @simonbyrne: I think that this is ready to go if you are happy with it.

**Pull request merged**